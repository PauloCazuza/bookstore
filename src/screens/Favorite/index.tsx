import { useState, useContext, useCallback } from "react";
import {
    VStack,
    Text,
    HStack,
} from "native-base";
import Ionicons from 'react-native-vector-icons/Ionicons';
import FormSearch, { FormDataProps } from "../../components/FormSearch";
import Spinner from "../../components/Spinner";
import { BookCtx } from "../../contexts/SearchBooks";
import { LoadingCtx } from "../../contexts/Loading";
import { IFormatList } from "../../interfaces/FormatData";
import ListCards from "../../components/ListCards";
import { FavoriteCtx } from "../../contexts/Favorites";
import { useFocusEffect } from '@react-navigation/native';

export default function Favorties() {
    const { favorites, filterFavorites } = useContext(FavoriteCtx);
    const [listFavorites, setListFavorites] = useState<IFormatList[]>(favorites);
    const { loading } = useContext(LoadingCtx);
    const { searchSimpleBook } = useContext(BookCtx);
    const [form, setForm] = useState<FormDataProps>({ search: "" } as FormDataProps);

    useFocusEffect(
        useCallback(() => {
            setListFavorites(favorites);
        }, [favorites])
    );

    async function submit(dataControl: FormDataProps) {
        const data = filterFavorites(dataControl.search);

        setListFavorites(data);
    }
    return (
        <VStack bgColor="gray .300" flex={1} px={5} my={5}>
            <VStack flex={1}>
                <FormSearch handleClick={submit} />
            </VStack>

            {loading && favorites.length === 0 && < Spinner />}
            <VStack flex={11}>
                <ListCards
                    listItems={listFavorites}
                />
            </VStack>
        </VStack >
    )
}