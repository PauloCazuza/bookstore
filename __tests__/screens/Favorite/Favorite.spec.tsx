import { NativeBaseProvider } from "native-base";
import renderer from "react-test-renderer";
import Favorite from "../../../src/screens/Favorite";

it('renders correctly Favorite', () => {
    const tree = renderer.create(<NativeBaseProvider><Favorite key={"test"} /></NativeBaseProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});