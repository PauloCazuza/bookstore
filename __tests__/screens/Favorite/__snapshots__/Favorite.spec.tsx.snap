// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Favorite 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
